### DISCUSSION TITLE
ATProto integration ideas

### DISCUSSION THREAD

**Discussion by 1dom** [tinyBlogPub/microblog.pub#14] [Ideas]
The appeal for microblog.pub for me has been an interest in open social protocols (activity pub, webmentions, micropub etc.)

Going forward, it looks like ATProto is getting a lot of interest. I don't actually do or use any social stuff, so haven't used bluesky or similar yet.

From what I can find though, conceptually, I think microblog.pub could also serve as an atproto [data repo](https://atproto.com/specs/repository), making it a single user [PDS](https://atproto.com/guides/glossary#pds-personal-data-server).

Reading what's available of its spec is interesting, but I'm finding it hard to get to actual implementation code or technical low level descriptions for e.g. a PDS API.

Anyone else have any opinions, knowledge or experience with ATProto?

**Comment by nigini** [comment]
I am on the same page as you @1dom: I am highly interested in open social protocols. I agree that adding some ATProto support would be really cool. I think, though, we should do a major refactoring (v3???) on the repo to modularize each protocol before adding new ones. (I started discussion #16 about this specific point!)

Regarding your questions about the ATP, I recommend checking [Ryan's](https://github.com/snarfed) stuff, more specifically his ["Procol translator library"](https://github.com/snarfed/granary) and his [Bridge Project](https://github.com/snarfed/bridgy).

**Reply by 1dom** [reply]
Thanks, helpful. I was aware of bridgy but hadn't looked much at granary - interesting!
